app/services/renderer/products.rb

```diff
@@ -160,32 +163,18 @@ def self.xml(url_options, current_store, current_currency, products)
     channel << create_node("title", current_store.name)
     channel << create_node("link", current_store.url)
     channel << create_node("description", "Find out about new products first! Always be in the know when new products become available")
-    
+
     if defined?(current_store.default_locale) && !current_store.default_locale.nil?
       channel << create_node("language", current_store.default_locale.downcase)
     else
       channel << create_node("language", "en-us")
     end
 
     products = products.except(:limit, :offset)
-    products.each_with_index do |product, index|
-      if product.is_in_hide_from_nav_taxon?
-        next
-      elsif product.feed_active?
-        if product.variants_and_option_values(current_currency).any?
-          product.variants.each do |variant|
-            if variant.show_in_product_feed?
-              channel << (item = create_node("item"))
-              complex_product(url_options, current_store, current_currency, item, product, variant)
-            end
-          end
-        else
-          channel << (item = create_node("item"))
-          basic_product(url_options, current_store, current_currency, item, product)
-        end
-      end
-
-      GC.start if index % 100 == 0 # forcing garbage collection
+     products.find_in_batches(batch_size: 100) do |products|
+      product_ids = products.pluck(:id)
```

Comment:
why pluck by Ids pass direct products.

Reply:
beacuse when i pass whole object by default it convert with to_s and look like #